rust/src/io.rs

```diff
@@ -29,13 +29,13 @@ pub mod object_reader;
 pub mod object_store;
 pub mod object_writer;
 pub mod reader;
+mod writer;
```

Comment:
should this be pub mod?

Reply:
was thining just re-export writer/reader via io, and make writer/reader to be private.

it takes quite some typing to import `lance::io::reader::FileReader` ? wdyt

Reply:
fine by me. was just curious why reader and writer were different

Reply:
was thinking to make `reader` the same as `writer` later.